pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -88,11 +88,18 @@ func LoadBalancerSrcRanges() []string {
 
 // GetLoadBalancer is an implementation of LoadBalancer.GetLoadBalancer
 func (gce *GCECloud) GetLoadBalancer(clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
-	loadBalancerName := cloudprovider.GetLoadBalancerName(service)
+	// For backword compatibility we had to fetch the loadBalancerName this way or else a simple assignment for service.Status.LoadBalancer.Name would be enough.
+	var loadBalancerName string
+	if len(service.Status.LoadBalancer.Name) > 0 {
+		loadBalancerName = service.Status.LoadBalancer.Name
```

Comment:
What happens if we lose Status?  Are we still trying to design for that @thockin ?  I think one option on AWS would be to tag the ELB with the service UID, so that we could name it whatever we want.  I don't think GCE has tagging support on these objects though